src/app/containers/CpsRecommendations/index.jsx

```diff
@@ -0,0 +1,203 @@
+`;
+
+const SectionLabelWrapper = styled.div`
+  padding: ${GEL_SPACING_DBL};
+`;
+
+const WsojWrapper = styled.div`
+  background-color: ${C_LUNAR};
+  padding-bottom: ${GEL_SPACING};
+`;
+
+const StyledStoryPromoLi = styled(StoryPromoLi)`
+  /* Override various paddings set by StoryPromoLi */
+  margin: -${GEL_SPACING_DBL_MINUS_HLF} 0;
+`;
+
+const getPromoItemProps = item => ({
+  src: path(['indexImage', 'path'], item),
+  href: path(['locators', 'assetUri'], item),
+  headline: path(['headlines', 'headline'], item),
+  alt: path(['indexImage', 'altText'], item),
+});
+
+const toStoryPromoItem = ({ assetUri, shortHeadline, imageHref }) => ({
+  headlines: {
+    headline: shortHeadline,
+  },
+  locators: {
+    assetUri,
+  },
+  indexImage: {
+    path: imageHref,
+    altText: 'Image Alt text', // This value would be changed as it's on our list of improvements.
+  },
+  uri: assetUri,
+});
+
+const transformItems = items => items.map(toStoryPromoItem);
+
+const CpsRecommendations = ({ parentColumns, items }) => {
+  const { script, service, dir, recommendations } = useContext(ServiceContext);
+  const { enabled } = useToggle('cpsRecommendations');
+
+  const { hasStoryRecommendations } = recommendations;
+
+  if (!hasStoryRecommendations || !enabled) return null;
+
+  const singleTransform = item => {
+    const { src, href, headline, alt } = getPromoItemProps(item);
+
+    const Img = <Image alt={alt} src={src} />;
+
+    const Info = (
+      <Headline script={script} service={service} promoHasImage as="span">
+        <Link href={href}>{headline}</Link>
+      </Headline>
+    );
+
+    return (
+      <Grid
+        columns={{
+          group0: 1,
+          group1: 1,
+          group2: 1,
+          group3: 1,
+          group4: 1,
+          group5: 1,
+        }}
+        enableGelGutters
+        dir={dir}
+      >
+        <StoryPromoWrapper image={Img} info={Info} />
+      </Grid>
+    );
+  };
+
+  const listTransform = promoItems => (
+    <Grid
+      columns={{
+        group0: 6,
+        group1: 6,
+        group2: 6,
+        group3: 6,
+        group4: 6,
+        group5: 6,
+      }}
+      as={StoryPromoUl}
+      enableGelGutters
+      dir={dir}
+    >
+      {promoItems.map(item => (
+        <Grid
+          item
+          columns={{
+            group0: 6,
+            group1: 6,
+            group2: 6,
+            group3: 6,
+            group4: 6,
+            group5: 6,
+          }}
+          as={StyledStoryPromoLi}
+          key={item.uri}
+          dir={dir}
+        >
+          {singleTransform(item)}
+        </Grid>
+      ))}
+    </Grid>
+  );
+
+  const renderSectionLabel = () => (
+    <SectionLabelWrapper>
+      <CustomSectionLabel
+        script={script}
+        service={service}
+        dir={dir}
+        id="recommendations-heading"
+        bar={false}
+        as="strong"
+      >
+        Recommendations
+      </CustomSectionLabel>
+    </SectionLabelWrapper>
+  );
```

Comment:
Maybe should be extracted if used elsewhere?

Reply:
https://github.com/bbc/simorgh/pull/6994#issuecomment-650163355